scripts/build/builders/builder.py

```diff
@@ -42,19 +43,51 @@ def generate(self):
     raise NotImplementedError()
 
   @abstractmethod
-  def build(self):
+  def do_build(self):
     """Perform an actual build"""
     raise NotImplementedError()
 
+  def do_generate_flashbundle(self):
```

Comment:
it is not immediately obvious that `do_*` methods are internal. Consider using one `_` to mark the internal methods as such.